File: healthone/lib/study_habits_folder/study_folder/activity.dart
class Activity {
  final int id;
  final String classCode;
  final String semester;
  final int duration;

  Activity({
    required this.id,
    required this.classCode,
    required this.semester,
    required this.duration,
  });

  factory Activity.fromMap(Map studyClassMap) {
    return Activity(
      id: studyClassMap['id'],
      classCode: studyClassMap['classCode'],
      semester: studyClassMap['semester'],
      duration: studyClassMap['duration'],
    );
  }
}

File: ExerciseAndMeditationMicroService/ExerciseAndMeditation/src/main/java/com/example/Model/Activity.java
package com.example.Model;

import lombok.Data;
import lombok.RequiredArgsConstructor;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;

import java.util.Date;

@Data
@Entity
@RequiredArgsConstructor
@Table(name = "Activity")
public class Activity {

    @Id
    @GeneratedValue(strategy=GenerationType.IDENTITY)
    private long id;

    private String classCode;
    private String semester;
    private long duration;

    public Activity(String classCode, String semester, long duration){
        this.classCode = classCode;
        this.semester = semester;
        this.duration = duration;
    }
}
